java/serving/src/main/java/feast/serving/service/OnlineServingServiceV2.java

```diff
@@ -110,13 +112,33 @@ public GetFeastServingInfoResponse getFeastServingInfo(
       throw new RuntimeException("Requested features list must not be empty");
     }
 
+    List<EntityProto.Entity> entities = this.registryRepository.getEntities();
+    Map<String, List<String>> entityNamesPerFeatureView =
+        retrievedFeatureReferences.stream()
+            .map(this.registryRepository::getFeatureViewSpec)
```

Comment:
so here we calling `getFeatureViewSpec` for each feature, which is not optimal, especially when there could be dozens features per single view. I think extracting feature view names -> deduplicating them (`distinct`) -> retrieving spec from registry would be much better.

Reply:
Good catch!